### ISSUE TITLE
[IMPROVEMENT] Fix outdated environment check in longhorn manager

### ISSUE THREAD

**Issue by innobead** [longhorn/longhorn#7571]
## Is your improvement request related to a feature? Please describe (👍 if you like this request)

Besides fixing the environment check in the bootstrap of the Longhorn manager, an environment check should also be added to the pre-upgrade job when installing Longhorn via Helm.

Consider consolidating this part together with the upgrade version check. 

[/app/daemon.go#L236-L244](https://github.com/longhorn/longhorn-manager/blob/a601b9b14da2ef3bb72f225dc3bc6b94515f0d33/app/daemon.go#L236-L244)
```go
func environmentCheck() error {
	namespaces := []lhtypes.Namespace{lhtypes.NamespaceMnt, lhtypes.NamespaceNet}
	nsexec, err := lhns.NewNamespaceExecutor(iscsiutil.ISCSIdProcess, lhtypes.HostProcDirectory, namespaces)
	if err != nil {
		return err
	}

	return iscsi.CheckForInitiatorExistence(nsexec)
}
```

**Comment by roger-ryao** [comment]
**Verified on master-head 20240711**

- longhorn master-head https://github.com/longhorn/longhorn/commit/69739d8a63231934c5f7bb0c70619130be25fe23


**The test steps**
https://github.com/longhorn/longhorn/issues/7571#issuecomment-2081844335

**Result Passed**
I set up a **Debian** cluster. After installing Longhorn, the status of `multipathd` and required packages will be recorded in the node's `Status.Conditions`.

```console
> k -n longhorn-system describe nodes.longhorn.io ryao-test-w2-d0ff0ce9-4lw25
Status:
  Auto Evicting:  false
  Conditions:
    Last Probe Time:       
    Last Transition Time:  2024-07-12T03:33:36Z
    Message:               Node ryao-test-w2-d0ff0ce9-4lw25 is ready
    Reason:                
    Status:                True
    Type:                  Ready
    Last Probe Time:       
    Last Transition Time:  2024-07-12T03:33:25Z
    Message:               
    Reason:                
    Status:                True
    Type:                  Schedulable
    Last Probe Time:       
    Last Transition Time:  2024-07-12T03:33:56Z
    Message:               
    Reason:                
    Status:                True
    Type:                  MountPropagation
    Last Probe Time:       
    Last Transition Time:  2024-07-12T03:33:56Z
    Message:               Missing packages: [nfs-common open-iscsi cryptsetup dmsetup]
    Reason:                PackagesNotInstalled
    Status:                False
    Type:                  RequiredPackages


```

Hi @mantissahz 
According to our chart, the number of Longhorn installations on **Ubuntu** clusters accounts for **56%** of the total. I think https://github.com/longhorn/longhorn-manager/blob/master/controller/node_controller.go#L1020-L1021 should also include checks for **Ubuntu**, to see if "nfs-common", "open-iscsi", "cryptsetup", and "dmsetup" are installed. This would be more convenient for us. WDYT?

**Comment by mantissahz** [comment]
>According to our chart, the number of Longhorn installations on Ubuntu clusters accounts for 56% of the total. I think https://github.com/longhorn/longhorn-manager/blob/master/controller/node_controller.go#L1020-L1021 should also include checks for Ubuntu, to see if nfs-common, open-iscsi, cryptsetup, and dmsetup are installed. This would be more convenient for us. WDYT?

We do include the check for `Ubuntu` just as `Debian`, they are the same.
https://github.com/longhorn/longhorn-manager/blob/master/controller/node_controller.go#L1018-L1021, line 1018 is including `ubuntu`

**Comment by roger-ryao** [comment]
**Verified on master-head 20240712**

- longhorn master-head https://github.com/longhorn/longhorn/commit/11066751cf818ddd93a4df0c457004e0fc95894c
- longhorn-manager master-head https://github.com/longhorn/longhorn-manager/commit/164df29644e1464ce1250ee5f2b99a8272a5851f

**Verified on v1.7.x-head 20240712**

- longhorn v1.7.x-head https://github.com/longhorn/longhorn/commit/bb7dc982080da42ff3a6055b852f0149c204a678
- longhorn-manager v1.7.x-head https://github.com/longhorn/longhorn-manager/commit/a805ee30f2cf48842e2dbcbece440ee9d2328d52


**The test steps**

https://github.com/longhorn/longhorn/issues/7571#issuecomment-2081844335


**Result Passed**


```console
curl -sSfL https://raw.githubusercontent.com/longhorn/longhorn/master/scripts/environment_check.sh | bash
[INFO]  Required dependencies 'kubectl jq mktemp sort printf' are installed.
[INFO]  All nodes have unique hostnames.
[INFO]  Waiting for longhorn-environment-check pods to become ready (0/2)...
[INFO]  All longhorn-environment-check pods are ready (2/2).
[INFO]  MountPropagation is enabled
[INFO]  Checking kernel release...
[INFO]  Checking iscsid...
[INFO]  Checking multipathd...
[INFO]  Checking packages...
[ERROR] cryptsetup is not found in ryao-test-w1-db48f7f9-rttw2.
[INFO]  Checking nfs client...
[INFO]  Cleaning up longhorn-environment-check pods...
[INFO]  Cleanup completed.


> kubectl -n longhorn-system describe nodes.longhorn.io ryao-test-w1-db48f7f9-rttw2 
Name:         ryao-test-w1-db48f7f9-rttw2
Status:
  Auto Evicting:  false
  Conditions:
    Last Probe Time:       
    Last Transition Time:  2024-07-12T11:35:24Z
    Message:               Node ryao-test-w1-db48f7f9-rttw2 is ready
    Reason:                
    Status:                True
    Type:                  Ready
    Last Probe Time:       
    Last Transition Time:  2024-07-12T11:35:24Z
    Message:               
    Reason:                
    Status:                True
    Type:                  Schedulable
    Last Probe Time:       
    Last Transition Time:  2024-07-12T11:35:54Z
    Message:               
    Reason:                
    Status:                True
    Type:                  MountPropagation
    Last Probe Time:       
    Last Transition Time:  2024-07-12T11:35:54Z
    Message:               Missing packages: [cryptsetup]
    Reason:                PackagesNotInstalled
    Status:                False
    Type:                  RequiredPackages



```

**Comment by derekbit** [comment]
Need to update the oficial doc

cc @mantissahz  @longhorn/qa

**Comment by derekbit** [comment]
@roger-ryao Please help check doc

**Comment by roger-ryao** [comment]
**Verified on master-head 20240726**

- website master-head https://github.com/longhorn/website/commit/6ed2582f9c7e5ba68c187986ba5f2aea1aca4fe3

**Result Passed**

require/doc : https://github.com/longhorn/website/pull/947

**Comment by roger-ryao** [comment]
Hi @derekbit @mantissahz 
     Does this fix plan to backport to `v1.6.3`?

**Comment by derekbit** [comment]
> Hi @derekbit @mantissahz Does this fix plan to backport to `v1.6.3`?

I am fine with the proposal. WDYT? @innobead
